e2e/actions/l2_challenger_test.go

```diff
@@ -197,3 +197,181 @@ interaction:
+	// outputOnL1, err := outputOracleContract.GetL2OutputAfter(nil, new(big.Int).SetUint64(block.Number))
+	blockNum := big.NewInt(int64(testdata.TargetBlockNumber))
+	outputIndex, err := outputOracleContract.GetL2OutputIndexAfter(nil, blockNum)
+	require.NoError(t, err)
+	outputOnL1, err := outputOracleContract.GetL2OutputAfter(nil, blockNum)
+	require.NoError(t, err)
+	block, err := propEngine.EthClient().BlockByNumber(t.Ctx(), blockNum)
+	require.NoError(t, err)
+	require.Less(t, block.Time(), outputOnL1.Timestamp.Uint64(), "output is registered with L1 timestamp of L2 tx output submission, past L2 block")
+	outputComputed, err := challengerRollupClient.OutputAtBlock(t.Ctx(), blockNum.Uint64())
+	require.NoError(t, err)
+	require.NotEqual(t, eth.Bytes32(outputOnL1.OutputRoot), outputComputed.OutputRoot, "output roots must different")
+
+	// deposit bond for challenger
+	challenger.ActDeposit(t, 1_000)
+	includeL1Block(t, miner, challenger.address)
+
+	// submit create challenge tx
+	txHash := challenger.ActCreateChallenge(t, outputIndex)
+
+	// include tx on L1
+	includeL1Block(t, miner, challenger.address)
+
+	// Check whether the submission was successful
+	receipt, err := miner.EthClient().TransactionReceipt(t.Ctx(), txHash)
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "failed to create challenge")
+
+	// check challenge created
+	colosseumContract, err := bindings.NewColosseum(sd.DeploymentsL1.ColosseumProxy, miner.EthClient())
+	require.NoError(t, err)
+	challenge, err := colosseumContract.GetChallenge(nil, outputIndex)
+	require.NoError(t, err)
+	require.NotNil(t, challenge, "challenge not found")
+
+interaction:
+	for {
+		status, err := colosseumContract.GetStatus(nil, outputIndex)
+		require.NoError(t, err)
+
+		switch status {
+		case chal.StatusChallengerTurn:
+			// call bisect by challenger
+			txHash = challenger.ActBisect(t, outputIndex)
+			includeL1Block(t, miner, challenger.address)
+		case chal.StatusAsserterTurn:
+			// call bisect by validator
+			txHash = validator.ActBisect(t, outputIndex)
+			includeL1Block(t, miner, validator.address)
+		case chal.StatusAsserterTimeout, chal.StatusReadyToProve:
+			txHash = challenger.ActProveFault(t, outputIndex)
+			includeL1Block(t, miner, challenger.address)
+		case chal.StatusProven:
+			// validate l2 output submitted by challenger
+			outputBlockNum := outputOnL1.L2BlockNumber.Uint64()
+			output := challenger.ActOutputAtBlockSafe(t, outputBlockNum)
+			isValid := guardian.ActValidateL2Output(t, output.OutputRoot, outputBlockNum)
+			require.False(t, isValid)
+			break interaction
+		default:
+			break interaction
+		}
+
+		// Check whether the submission was successful
+		receipt, err = miner.EthClient().TransactionReceipt(t.Ctx(), txHash)
+		require.NoError(t, err)
+		require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "failed to progress interactive proof")
+	}
+
+	// Check the status of challenge is StatusApproved(7)
```

Comment:
I think it should be StatusProven.
```suggestion
	// Check the status of challenge is StatusProven(6)
```